ext/har/logger_test.go

```diff
@@ -0,0 +1,268 @@
+func createTestProxy(logger *Logger) *httptest.Server {
+    proxy := goproxy.NewProxyHttpServer()
+    proxy.OnRequest().DoFunc(logger.OnRequest)
+    proxy.OnResponse().DoFunc(logger.OnResponse)
+    return httptest.NewServer(proxy)
+}
+
+// createProxyClient creates an HTTP client that uses the given proxy
+func createProxyClient(proxyURL string) *http.Client {
+    proxyURLParsed, _ := url.Parse(proxyURL)
+    tr := &http.Transport{
+        Proxy: http.ProxyURL(proxyURLParsed),
+    }
+    return &http.Client{Transport: tr}
+}
+
+func TestHarLoggerBasicFunctionality(t *testing.T) {
+    testCases := []struct {
+        name           string
+        method         string
+        body           string
+        contentType    string
+        expectedMethod string
+    }{
+        {
+            name:           "GET Request",
+            method:         http.MethodGet,
+            expectedMethod: http.MethodGet,
+        },
+        {
+            name:           "POST Request",
+            method:         http.MethodPost,
+            body:           `{"test":"data"}`,
+            contentType:    "application/json",
+            expectedMethod: http.MethodPost,
+        },
+    }
+
+    for _, tc := range testCases {
+        t.Run(tc.name, func(t *testing.T) {
+            background := httptest.NewServer(ConstantHandler("hello world"))
+            defer background.Close()
+
+            var exportedEntries []Entry
+            exportFunc := func(entries []Entry) {
+                exportedEntries = append(exportedEntries, entries...)
+            }
+            logger := NewLogger(exportFunc)
+            defer logger.Stop()
+
+            proxyServer := createTestProxy(logger)
+            defer proxyServer.Close()
+
+            client := createProxyClient(proxyServer.URL)
+
+            // Prepare request
+            req, err := http.NewRequest(tc.method, background.URL, strings.NewReader(tc.body))
+            require.NoError(t, err, "Should create request")
+            if tc.contentType != "" {
+                req.Header.Set("Content-Type", tc.contentType)
+            }
+
+            // Send request and capture response
+            resp, err := client.Do(req)
+            require.NoError(t, err, "Should send request successfully")
+            defer resp.Body.Close()
+
+            // Read response body
+            bodyBytes, _ := io.ReadAll(resp.Body)
+            body := string(bodyBytes)
+            assert.Equal(t, "hello world", body, "Response body should match")
+
+            time.Sleep(200 * time.Millisecond)
+
+            // Verify HAR entry
+            entries := logger.GetEntries()
+            require.Len(t, entries, 1, "Should have one log entry")
+            entry := entries[0]
+            assert.Equal(t, tc.expectedMethod, entry.Request.Method, "Request method should match")
+
+            // Verify exported entries
+            assert.Len(t, exportedEntries, 0, "Should not have exported entries yet")
```

Comment:
Big NO, don't call GetEntries, this will lead to flaky tests.
Instead, you can add the test entries directly inside the callback handler, and it will be deterministic.